### PR TITLE
Fixed bug KAA-496

### PR DESCRIPTION
**Pull request by RostakaGmfun**
Now client registers again if it receives `KAATCP_CONNACK_REFUSE_BAD_CREDENTIALS` from server.

### CODE REVIEW

File: client/client-multi/client-c/src/kaa/platform-impl/kaa_tcp_channel.c [hunk 1]
```diff
@@ -949,6 +954,12 @@ void kaa_tcp_channel_connack_message_callback(void *context, kaatcp_connack_t me
                 channel->keepalive.last_sent_keepalive = channel->keepalive.last_receive_keepalive;
             }
 
+        } else if(message.return_code == (uint16_t) KAATCP_CONNACK_REFUSE_BAD_CREDENTIALS) {
```
Comment (ghost) [inline client/client-multi/client-c/src/kaa/platform-impl/kaa_tcp_channel.c]: Please, add code style space after 'if'

File: client/client-multi/client-c/src/kaa/platform-impl/kaa_tcp_channel.c [hunk 2]
```diff
@@ -113,7 +114,11 @@ typedef struct {
     kaa_tcp_encrypt_t              encryption;
 } kaa_tcp_channel_t;
 
+extern kaa_error_t kaa_status_set_registered(kaa_status_t *self, bool is_registered);
 
+typedef struct kaa_status_holder_t {
```
Comment (Karnil) [inline client/client-multi/client-c/src/kaa/platform-impl/kaa_tcp_channel.c]: it's not clear where you use this structure.

**Comment by ghost** [comment]
LGTM

**Comment by Karnil** [comment]
it seems ok

**Pull request merged**